Fix issues with path resolving and listing files

volodymyrpoli opened the pull request.

volodymyrpoli committed:
Fix paths issues (double slashes, end slashes)

volodymyrpoli committed:
Fix recursive listing of files

volodymyrpoli committed:
Fix overwrite parameter conversion

volodymyrpoli committed:
Replace maxAmountOfFiles with maxDepth, as it is a more expected result

When we read files by their number, we never get the right result.
For example, we have the following file system structure:

folder1/
- subfolder1/
--- file1
--- file2
- file3
- file4
folder2/
- subfolder2/
--- file5
--- file6
- file5

and maxAmountOfFiles = 3, then we will read [folder1, subfolder1,
file3, file3, folder2, subfolder2, file5]. And this is not expected
result. It is better to read by depth, which will give the expected
result.

volodymyrpoli committed:
Update package name and version

xaviered commented on README.md:
````diff
@@ -12,7 +12,7 @@ php-WebHDFS is a PHP client for [WebHDFS](http://hadoop.apache.org/docs/r2.0.3-a
 
 ```bash
-composer require simpleenergy/php-webhdfs
+composer require dreamfactory/php-webhdfs
````
Changes look good, but I think we should still keep the original package name. I'll be glad to merge your changes as a new version if you remove this.

xaviered commented on composer.json:
```diff
@@ -1,9 +1,15 @@
 {
-  "name": "simpleenergy/php-webhdfs",
+  "name": "dreamfactory/php-webhdfs",
```
and of course this one too

xaviered commented on composer.json:
```diff
@@ -1,9 +1,15 @@
+  "name": "dreamfactory/php-webhdfs",
   "description": "PHP WebHDFS, forked from https://github.com/simpleenergy/php-WebHDFS",
   "minimum-stability": "stable",
+  "repositories": [
```
same here

daniilly committed:
[BUGFIX] DP-460

- Get rid of invalid 'CURLINFO_EFFECTIVE_URL' cURL option
- Get rid of improper 'Content-Length' HTTP header
- Remove version composer`s property, as it should be
  omitted relying on the documentation https://getcomposer.org/doc/04-schema.md#version
